dotnet/src/Extensions/PromptTemplate.Handlebars/Helpers/SystemHelpers/KernelSystemHelpers.cs

```diff
@@ -0,0 +1,169 @@
+            }
+
+            writer.Write($"<{parameters["role"]}~>", false);
+            options.Template(writer, context);
+            writer.Write($"</{parameters["role"]}~>", false);
+        });
+
+        handlebarsInstance.RegisterHelper("set", (writer, context, arguments) =>
+        {
+            var name = string.Empty;
+            object value = string.Empty;
+            if (arguments[0].GetType() == typeof(HashParameterDictionary))
+            {
+                // Get the parameters from the template arguments
+                var parameters = arguments[0] as IDictionary<string, object>;
+                name = (string)parameters!["name"];
+                value = parameters!["value"];
+            }
+            else
+            {
+                name = arguments[0].ToString();
+                value = arguments[1];
+            }
+
+            if (variables.TryGetValue(name, out var argVal))
+            {
+                variables[name] = value;
+            }
+            else
+            {
+                variables.Add(name, value);
+            }
+        });
+
+        handlebarsInstance.RegisterHelper("get", (in HelperOptions options, in Context context, in Arguments arguments) =>
+        {
+            if (arguments[0].GetType() == typeof(HashParameterDictionary))
+            {
+                var parameters = arguments[0] as IDictionary<string, object>;
+                return variables[(string)parameters!["name"]];
+            }
+
+            return variables[arguments[0].ToString()];
+        });
+
+        handlebarsInstance.RegisterHelper("json", (in HelperOptions options, in Context context, in Arguments arguments) =>
+        {
+            object objectToSerialize = arguments[0];
+            string json = objectToSerialize.GetType() == typeof(string) ? (string)objectToSerialize : JsonSerializer.Serialize(objectToSerialize);
+
+            return json;
+        });
+
+        handlebarsInstance.RegisterHelper("concat", (in HelperOptions options, in Context context, in Arguments arguments) =>
+        {
+            return string.Concat(arguments.Select(var => var?.ToString()));
+        });
+
+        handlebarsInstance.RegisterHelper("raw", (writer, options, context, arguments) =>
+        {
+            options.Template(writer, null);
+        });
+
+        handlebarsInstance.RegisterHelper("array", (in HelperOptions options, in Context context, in Arguments arguments) =>
+        {
+            // Convert all the arguments to an array
+            var array = arguments.Select(a => a).ToList();
+
+            return array;
+        });
+
+        handlebarsInstance.RegisterHelper("range", (in HelperOptions options, in Context context, in Arguments arguments) =>
+        {
+            var start = int.Parse(arguments[0].ToString(), CultureInfo.InvariantCulture);
+            var end = int.Parse(arguments[1].ToString(), CultureInfo.InvariantCulture) + 1;
```

Comment:
Should this be using the CultureInfo from the Kernel? Or is the Kernel not available here?

Reply:
Drilled down the kernel to use `kernel.Culture`